src/main/java/uk/gov/hmcts/cmc/claimstore/services/search/CaseRepository.java

```diff
@@ -29,5 +31,9 @@ void saveDefendantResponse(Claim claim, String defendantId, String defendantEmai
     Optional<Claim> getByLetterHolderId(String id, String authorisation);
 
     void requestMoreTimeForResponse(String authorisation, Claim claim, LocalDate newResponseDeadline);
+
+    void updateSettlement(Claim claim, Settlement settlement, String authorisation, CaseEvent event);
+
+    void reachSettlementAgreement(Claim claim, Settlement settlement, String authorisation, CaseEvent event);
```

Comment:
`CaseEvent` is CCD related, please move this down to a CCD related implementation.